### PR TITLE
fix(PEcAn.SIPNET): return empty dataframe when date filtering removes all segments (#4007)

### LINKED ISSUE #4007: segment_dataframe: return empty if no segments after filtering
Spotted by @divine7022 in https://github.com/PecanProject/pecan/pull/4006#discussion_r3284509836:

> we could end up with 1 row segments frame where start_date is set but every other column (end_date, crop_code, site_id, pft) is NA; and that gets returned to write_segment_configs, which then loops over  row and tries to build segment configs with NA dates and NA pft. downstream error site is far from the actual problem

Since we already return an empty dataframe if there are zero rows _before_ filtering (row 115 in the version we're discussing today) , it would be consistent to replace this error with a simple `return(segments)` -- possibly with a warning  that no relevant segments were found? Today I lean towards just returning silently (If no segments are relevant to the requested time range, we can simply run the existing unsegmented config and it's not a problem), but am open to counterarguments.

### PR DESCRIPTION
**Pull request by ANAMASGARD**
## Description

Fixes a bug in `segment_dataframe()` where date filtering could leave zero valid crop-cycle segments, but the function still continued and executed `segments[1, "start_date"] <- run_start` on an empty dataframe. In R, that assignment creates a **ghost 1-row frame** with `start_date` set and all other columns (`end_date`, `crop_code`, `site_id`, `pft`) as `NA`.

That ghost row was passed to `write_segment_configs()`, which then tried to build segment configs with invalid dates and PFTs. The failure appeared far downstream from the actual root cause.

**Change:** When filtering removes all segments, return the empty dataframe immediately (silent return, consistent with the existing pre-filter check at line 135). Downstream `write_segment_configs()` already handles `nrow(segments) == 0` by returning the unaltered `job.sh`.

**Files changed:**
- `models/sipnet/R/write_segmented_configs.R` — early return after filtering
- `models/sipnet/tests/testthat/test-write_segmented_configs.R` — unit + integration tests
- `CHANGELOG.md` — Unreleased / Fixed entry
- `models/sipnet/NEWS.md` — package news entry

## Motivation and Context

Closes #4007.

This edge case was spotted by @divine7022 during review of #4006. When `run$start.date` falls after all crop-cycle dates in the schedule, no segments are relevant to the requested time range. The correct behavior is to run the existing unsegmented config — not to error downstream with NA-filled segment rows.

Related: #4006

## Review Time Estimate

- [x] Within one week

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue) — #4007
## Checklist

- [x] I agree that PEcAn Project may distribute my contribution under any or all of
  - the same license as the existing code,
  - and/or the BSD 3-clause license.
- [x] I have updated the CHANGELOG.md.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Test Result
All tests passed.
image

That means no broken ghost row — fix is working. (Below ScreenShot)
image

Ran all SIPNET tests
image

### CODE REVIEW

File: models/sipnet/tests/testthat/test-write_segmented_configs.R [hunk 1]
```diff
@@ -98,3 +98,99 @@ test_that("write_segmented_configs", {
...
+      inputs = list(crop_changes = list(path = crp_chg_path)),
+      start.date = "2025-01-10",
+      end.date = "2025-01-20"
+    )
+  ))
+
+  result <- PEcAn.SIPNET:::segment_dataframe(run_settings)
+
+  expect_equal(nrow(result), 0)
+  expect_gt(ncol(result), 0)
```
Comment (dlebauer) [inline models/sipnet/tests/testthat/test-write_segmented_configs.R]: can we make this more explicit by naming the required columns?

```suggestion
  expect_named(
    result, 
    c("start_date", "crop_code", "end_date")
  )
```

**Review by dlebauer — APPROVED** [review]
Thanks for this fix - it fixes #4007 and adds new tests. Ready to merge.

I made an optional suggestion - to test for specific col names rather than that ncol > 0.

File: models/sipnet/R/write_segmented_configs.R [hunk 2]
```diff
@@ -160,11 +160,8 @@ segment_dataframe <- function(run_settings) {
...
         .data$end_date > .env$run_start
       )
-    if (nrow(segments) < 1) {
-      PEcAn.logger::logger.error(
-        "Filtering resulted in no segments. ",
-        "This is an invalid state; check settings and events.json."
-      )
+    if (nrow(segments) == 0) {
+      return(segments)
     }
```
Comment (divine7022) [inline models/sipnet/R/write_segmented_configs.R]: I still think that's the right behavior for the normal "no crop changes fall within the run window" case. The only thing I am wondering about is that this same branch also fires if a csv gets wired to the wrong site, so the run just goes unsegmented with no trace. Would it be worth adding a simple logger.info here (rather than a warning) just to leave a trace

```suggestion
    if (nrow(segments) == 0) {
    PEcAn.logger::logger.info(
      "no crop cycle segments fall within the run window; running unsegmented"
    )
    return(segments)
  }
```
totally optional, if you would rather keep it fully silent, that's reasonable. Just thought a silent no-op could make a real misconfiguration a little harder to spot

**Review by divine7022 — APPROVED** [review]
lgtm!, one small thing and agree with david's inline

File: models/sipnet/R/write_segmented_configs.R (hunk 2 above)
Comment (infotroph) [reply models/sipnet/R/write_segmented_configs.R]: I agree this deserves some kind of warning -- if we reach this line it means we had a site file with events in it (missing/empty files get caught by the earlier return) but they all fell outside the simulation window. My gut says that's technically possible but much more likely to be a mistake / bad data than to be intended.
